phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java

```diff
@@ -45,7 +45,9 @@ public ResultSet executeQuery() throws SQLException {
 
     @Override
     public ResultSet getResultSet() throws SQLException {
-        return new LoggingPhoenixResultSet(super.getResultSet(), phoenixMetricsLog, sql);
+        ResultSet resultSet = super.getResultSet();
```

Comment:
nit: add a comment that call to `getResultSet()` is not idempotent. Hence we cache and return accordingly.